cmd/policy-assistant/pkg/matcher/traffic.go

```diff
@@ -71,10 +75,258 @@ func (p *TrafficPeer) IsExternal() bool {
+	var deploymentPeers []TrafficPeer
+	kubeClient, err := kube.NewKubernetesForContext("")
+	utils.DoOrDie(err)
+	kubeNamespaces, err := kubeClient.GetAllNamespaces()
+	if err != nil {
+		logrus.Fatalf("unable to read namespaces from kube: %+v", err)
+	}
+
+	for _, namespace := range kubeNamespaces.Items {
+		kubeDeployments, err := kubeClient.GetDeploymentsInNamespace(namespace.Name)
+		if err != nil {
+			logrus.Fatalf("unable to read deployments from kube, ns '%s': %+v", namespace.Name, err)
+		}
+		for _, deployment := range kubeDeployments {
+			tmpInternalPeer := InternalPeer{
+				Workload: namespace.Name + "/deployment/" + deployment.Name,
+			}
+			tmpPeer := TrafficPeer{
+				Internal: &tmpInternalPeer,
+			}
+			tmpPeerTranslated := tmpPeer.Translate()
+			if tmpPeerTranslated.Internal.Workload != "" {
+				deploymentPeers = append(deploymentPeers, tmpPeerTranslated)
+			}
+
+		}
+
+	}
+
+	return deploymentPeers
+}
+
+func DaemonSetsToTrafficPeers() []TrafficPeer {
+	var daemonSetPeers []TrafficPeer
+	kubeClient, err := kube.NewKubernetesForContext("")
+	utils.DoOrDie(err)
+	kubeNamespaces, err := kubeClient.GetAllNamespaces()
+	if err != nil {
+		logrus.Fatalf("unable to read namespaces from kube: %+v", err)
+	}
+
+	for _, namespace := range kubeNamespaces.Items {
+		kubeDaemonSets, err := kubeClient.GetDaemonSetsInNamespace(namespace.Name)
+		if err != nil {
+			logrus.Fatalf("unable to read daemonSets from kube, ns '%s': %+v", namespace.Name, err)
+		}
+		for _, daemonSet := range kubeDaemonSets {
+			tmpInternalPeer := InternalPeer{
+				Workload: namespace.Name + "/daemonset/" + daemonSet.Name,
+			}
+			tmpPeer := TrafficPeer{
+				Internal: &tmpInternalPeer,
+			}
+			tmpPeerTranslated := tmpPeer.Translate()
+			if tmpPeerTranslated.Internal.Workload != "" {
+				daemonSetPeers = append(daemonSetPeers, tmpPeerTranslated)
+			}
+		}
+
+	}
+
+	return daemonSetPeers
+}
+
+func StatefulSetsToTrafficPeers() []TrafficPeer {
+	var statefulSetPeers []TrafficPeer
+	kubeClient, err := kube.NewKubernetesForContext("")
+	utils.DoOrDie(err)
+	kubeNamespaces, err := kubeClient.GetAllNamespaces()
+	if err != nil {
+		logrus.Fatalf("unable to read namespaces from kube: %+v", err)
+	}
+
+	for _, namespace := range kubeNamespaces.Items {
+		kubeStatefulSets, err := kubeClient.GetStatefulSetsInNamespace(namespace.Name)
+		if err != nil {
+			logrus.Fatalf("unable to read statefulSets from kube, ns '%s': %+v", namespace.Name, err)
+		}
+		for _, statefulSet := range kubeStatefulSets {
+			tmpInternalPeer := InternalPeer{
+				Workload: namespace.Name + "/statefulset/" + statefulSet.Name,
+			}
+			tmpPeer := TrafficPeer{
+				Internal: &tmpInternalPeer,
+			}
+			tmpPeerTranslated := tmpPeer.Translate()
+			if tmpPeerTranslated.Internal.Workload != "" {
+				statefulSetPeers = append(statefulSetPeers, tmpPeerTranslated)
+			}
+		}
+
+	}
+
+	return statefulSetPeers
+}
+
+func ReplicaSetsToTrafficPeers() []TrafficPeer {
```

Comment:
nit: could you add comments describing the functions? At least for this one and `PodsToTrafficPeers()`. Including the fact that it ignores Pods associated with Deployments, etc.

Reply:
comments added